### ISSUE TITLE
Certificate confusion in /var/lib/boot2docker/ (1.9.0-rc2)

### ISSUE THREAD

**Issue by jamshid** [docker/machine#2111]
I installed docker toolbox 1.9.0-rc2 https://github.com/docker/toolbox/releases/tag/v1.9.0-rc2 on my OS X desktop (192.168.2.5). I am trying to connect to the docker vm (192.168.99.100:2376) via an ssh tunnel running on my notebook. 
```
ssh -nNT -L "2205:192.168.99.100:22" 192.168.2.5 &   # ssh proxy so I can copy the docker certs
mkdir -p ~/.docker-tls-desktop ; sshpass -p "tcuser" ssh -p 2205 docker@127.0.0.1 sudo tar -cf - -C /var/lib/boot2docker/tls ca.pem cert.pem key.pem | tar -xvf - -C ~/.docker-tls-desktop
ssh -nNT -L "23905:192.168.99.100:2376" 192.168.2.5 &  # docker port proxy
```
I then set these docker env variables to connect via the docker port proxy:
```
export DOCKER_HOST="tcp://127.0.0.1:23905"
export DOCKER_CERT_PATH=~/.docker-tls-desktop/
export DOCKER_TLS_VERIFY=1
```
But the docker client on my notebook still cannot talk to the docker server in the VM on my desktop:
```
$ docker ps
An error occurred trying to connect: Get https://127.0.0.1:23905/v1.20/containers/json: x509: certificate is valid for 192.168.99.100, not 127.0.0.1
```
Logging into the VM, according to tls/hostnames the docker cert should cover ip 127.0.0.1:
```
docker@default:~$ sudo cat /var/lib/boot2docker/tls/hostnames
default,127.0.0.1,10.0.2.15,192.168.99.100
```
but the certs in `tls` are *not actually used*. Instead, the ones in /var/lib/boot2docker/*.pem are used:
```
docker@default:~$ cat /var/lib/boot2docker/profile 
...
CACERT=/var/lib/boot2docker/ca.pem
SERVERKEY=/var/lib/boot2docker/server-key.pem
SERVERCERT=/var/lib/boot2docker/server.pem
```
Why?

Anyway, copying the three files (including renaming serverkey.pem to server-key.pem) from /var/lib/boot2docker/tls to /var/lib/boot2docker/ and "sudo /etc/init.d/docker restart" fixed the cert validation error.

PS: why is there both /var/lib/boot2docker/etc/hostname and /var/lib/boot2docker/tls/hostnames? Seems etc/hostname and the root level *.pem's should go away, use only the files in the tls directory.

PS2: if there's a better way to do this, let me know. Maybe I can manually(?) create a docker-machine configuration for this, but seems I still need a ssh tunnel to proxy the VM, unless I were to port-forward the VM's docker port on the host (my desktop) ip.

**Comment by dmp42** [comment]
ping @docker/machine-maintainers

**Comment by nathanleclaire** [comment]
Dupe (I think?) of: https://github.com/docker/machine/issues/2018

> PS2: if there's a better way to do this, let me know.

@jamshid, by "this" do you mean "forward the daemon port to localhost"?

**Comment by jamshid** [comment]
Yes, #2018 looks like the same issue, closing this one. Just to be clear, this means the the docker server cert will always include 127.0.0.1, not just the VM's ip address, right? Hopefully the fix will be in the 0.5.0 release pretty please? 

>@jamshid, by "this" do you mean "forward the daemon port to localhost"?

Yes, was just wondering if there's a better way to expose a VM's docker daemon port to another machine (that can only see the VM's host). Seems this is something Toolbox users would want, so they can control their docker environment from another machine on their network.

Maybe I should just configure VirtualBox to forward the docker port to the host machine's network interface.

I'm assuming I understand https://docs.docker.com/articles/https/ and that opening up the docker port is safe, since docker server simply won't talk to a client unless the client has a proper client cert.

**Comment by nathanleclaire** [comment]
> Yes, #2018 looks like the same issue, closing this one. Just to be clear, this means the the docker server cert will always include 127.0.0.1, not just the VM's ip address, right? Hopefully the fix will be in the 0.5.0 release pretty please?

I think you're referencing https://github.com/docker/machine/pull/2065 which should be released in 0.5.0.

**Comment by nathanleclaire** [comment]
SSH port forwarding and Virtualbox NAT port forwarding both seem reasonable solutions to me for exposing the daemon port on `localhost`.
